src/Conditionals.elm

```diff
@@ -0,0 +1,544 @@
+trafficLightActionCase : String -> String
+trafficLightActionCase color =
+    case color of
+        "Red" ->
+            "Stop"
+
+        "Yellow" ->
+            "Prepare to stop"
+
+        "Green" ->
+            "Go"
+
+        _ ->
+            color ++ "? Are we on Mars?"
+
+
+convertTo12HourFormat : Int -> ( Int, String )
+convertTo12HourFormat hour =
+    if hour >= 0 && hour <= 23 then
+        let
+            ( convertedHour, period ) =
+                if hour < 12 then
+                    if hour == 0 then
+                        ( 12, "AM" )
+
+                    else
+                        ( hour, "AM" )
+
+                else if hour == 12 then
+                    ( 12, "PM" )
+
+                else
+                    ( hour - 12, "PM" )
+        in
+        ( convertedHour, period )
+
+    else
+        ( -1, "Extra time? This isn't football, hun!" )
+
+
+weekdayMoodIfElse : String -> String
+weekdayMoodIfElse day =
+    if day == "Monday" then
+        "Back to the grind!"
+
+    else if day == "Tuesday" || day == "Wednesday" || day == "Thursday" then
+        "Just another day in the grind"
+
+    else if day == "Friday" then
+        "TGIF!"
+
+    else if day == "Saturday" || day == "Sunday" then
+        "It's the weekend, baby!"
+
+    else
+        "Is that a public holiday?"
+
+
+weeklydayMoodCase : String -> String
+weeklydayMoodCase day =
+    case day of
+        "Monday" ->
+            "Back to the grind!"
+
+        "Tuesday" ->
+            "Just another day in the grind"
+
+        "Wednesday" ->
+            "Just another day in the grind"
+
+        "Thursday" ->
+            "Just another day in the grind"
+
+        "Friday" ->
+            "TGIF!"
+
+        "Saturday" ->
+            "It's the weekend, baby!"
+
+        "Sunday" ->
+            "It's the weekend, baby!"
+
+        _ ->
+            "Is that a public holiday?"
+
+
+fizzBuzzGame : Int -> String
+fizzBuzzGame number =
+    case ( number // 3, number // 5 ) of
+        ( 0, 0 ) ->
+            "FizzBuzz"
+
+        ( 0, _ ) ->
+            "Fizz"
+
+        ( _, 0 ) ->
+            "Buzz"
+
+        _ ->
+            if number < 0 then
+                "Hmm, negative numbers and zero are not very fizzy or buzzy, are they?"
+
+            else if number > 100 then
+                String.fromInt number ++ " - Just another number in the universe!"
+
+            else
+                String.fromInt number
+
+
+checkValidDate : String -> Int -> String
+checkValidDate month day =
+    case month of
+        "January" ->
+            if day >= 1 && day <= 31 then
+                "Valid Date"
+
+            else
+                "Invalid Date: January has only 31 days"
+
+        "February" ->
+            if day >= 1 && day <= 28 then
+                "Valid Date"
+
+            else
+                "Invalid Date: February has only 28 days"
+
+        "March" ->
+            if day >= 1 && day <= 31 then
+                "Valid Date"
```

Comment:
Looks like you didn't implement my suggestion on this!

https://github.com/chelzdev/hamelin-assignments/pull/1#discussion_r1478824649

Reply:
> Looks like you didn't implement my suggestion on this!
> 
> [#1 (comment)](https://github.com/chelzdev/hamelin-assignments/pull/1#discussion_r1478824649)

I will implement it right away.
Thank you.